snuba/query/project_extension.py

```diff
@@ -112,9 +112,43 @@ def do_post_processing(
                 if len(exclude_group_ids) > max_group_ids_exclude:
                     query.set_final(True)
                 else:
-                    query.add_conditions(
-                        [(["assumeNotNull", ["group_id"]], "NOT IN", exclude_group_ids)]
-                    )
+                    # Check if there is a more selective conditions around group_id/issue
+                    # in the query conditions. Privilege the positive conditions instead of
+                    # the negative ones.
+                    strict_issue_conditions = False
+                    replaced_conditions: List[Condition] = []
+                    for c in query.get_conditions():
+                        new_condition = c
+                        if (
+                            util.is_condition(c)
+                            and c[0] in ["group_id", "issue"]
```

Comment:
this is gross. Cannot thing about a better way.

Reply:
I don't have a better idea either (at least one that doesn't require baking in alias awareness into the schema definition, and genericizing this to deal with `IN`/`NOT IN` clauses for any columns.)

This should probably at least comment the abstraction leak/dataset coupling here so we know it's not ideal/intentional rather than leaving it in a review comment.